sql/core/src/main/scala/org/apache/spark/sql/execution/columnar/GenerateColumnAccessor.scala

```diff
@@ -109,6 +115,14 @@ object GenerateColumnAccessor extends CodeGenerator[Seq[DataType], ColumnarItera
               rowWriter.write($index, (Decimal) null, $p, $s);
             }
            """
+        case CalendarIntervalType =>
+          // For CalendarInterval, it should have 16 bytes to store months(Int), days(Int),
+          // microseconds(Long) for future update even it's null now.
+          s"""
+            if (mutableRow.isNullAt($index)) {
+              rowWriter.write($index, (CalendarInterval) null);
+            }
+           """
```

Comment:
Is this patch work related to this discussion? https://github.com/apache/spark/pull/26680#discussion_r351153420
If we can remove this patch, the gen'd code looks nicer to me. cc: @cloud-fan

Reply:
After a second thought, I guess this patch is needless for CalendarInterval which is a `DirectCopyColumnType`